lib/chef/provider/aws_security_group.rb

```diff
@@ -94,13 +94,21 @@ def update_inbound_rules(sg, vpc)
     # Actually update the rules (remove, add)
     #
     update_rules(desired_rules, sg.ip_permissions,
-
       authorize: proc do |port_range, protocol, actors|
         names = actors.map { |a| a.is_a?(Hash) ? a[:group_id] : a }
         converge_by "authorize #{names.join(', ')} to send traffic to group #{new_resource.name} (#{sg.id}) on port_range #{port_range.inspect} with protocol #{protocol || 'nil'}" do
           names.each do |iprange|
            begin
             if iprange.include?('-')
+              # user_id_group_pairs allows to add inbound rules for source security group
+              sg.authorize_ingress({
+                ip_permissions: [{
+                  ip_protocol: protocol,
+                  from_port: port_range.first,
+                  to_port: port_range.last,
+                  user_id_group_pairs: actors
+                }]
+              })
```

Comment:
I am referring to this http://docs.aws.amazon.com/sdkforruby/api/Aws/EC2/SecurityGroup.html#authorize_egress-instance_method. Since there are many options which can be added depending on the rules and this will create more complexity if introduced as function since the ip_permissions set can be changed depending on the rules.